packages/core/src/render3/state.ts

```diff
@@ -119,29 +119,114 @@ export function getLView(): LView {
   return lView;
 }
 
+let activeHostDirectiveIndex: number|null = null;
 let activeHostContext: {}|null = null;
 let activeHostElementIndex: number|null = null;
+let activeHostDirectiveInheritanceDepth = 0;
 
 /**
  * Sets the active host context (the directive/component instance) and its host element index.
  *
  * @param host the directive/component instance
- * @param index the element index value for the host element where the directive/component instance
- * lives
+ * @param elementIndex the element index value for the host element where
+ *                     the directive/component instance lives
  */
-export function setActiveHost(host: {} | null, index: number | null = null) {
-  activeHostContext = host;
-  activeHostElementIndex = index;
-}
-
-export function getActiveHostContext() {
-  return activeHostContext;
+export function setActiveHost(
+    host: {} | null, elementIndex: number | null = null, directiveIndex: number | null = null) {
+  if (activeHostContext !== host) {
+    activeHostContext = host;
+    activeHostElementIndex = elementIndex;
+    activeHostDirectiveIndex = directiveIndex;
+    activeHostDirectiveInheritanceDepth = 0;
+  }
 }
 
+/**
+ * Returns the element index value of the current host element.
+ *
+ * Note that this is only active when `hostBinding` functions are being processed.
+ */
 export function getActiveHostElementIndex() {
   return activeHostElementIndex;
 }
 
+/**
+ * Returns the current computed index value of the current directive.
+ *
+ * Say for example we have an element that has two directives on it:
```

Comment:
```suggestion
 * For example we have an element that has two directives on it:
```
`Say` is unnecessary. Also all comments below.